pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -208,11 +208,16 @@ def on_save_checkpoint(self, trainer, pl_module) -> Dict[str, Any]:
             "best_model_score": self.best_model_score,
             "best_model_path": self.best_model_path,
             "current_score": self.current_score,
+            "dirpath": self.dirpath
         }
 
     def on_load_checkpoint(self, checkpointed_state: Dict[str, Any]):
         self.best_model_score = checkpointed_state["best_model_score"]
         self.best_model_path = checkpointed_state["best_model_path"]
+        dirpath = checkpointed_state.get("dirpath", self.dirpath)
+        # If the dirpath exists, checkpoints should be grouped together.
+        if dirpath is not None and os.path.exists(dirpath):
+            self.dirpath = dirpath
```

Comment:
what if dirpath in the new checkpoint instance is different?

Reply:
Ok, let's remove this one for now until we find a better design choice.